selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/theia/TheiaBuildPluginTest.java

```diff
@@ -108,10 +110,29 @@ public void pluginShouldBeBuilt() {
     theiaTerminal.waitTab(WS_THEIA_IDE_TERMINAL_TITLE);
     theiaTerminal.clickOnTab(WS_THEIA_IDE_TERMINAL_TITLE);
     theiaTerminal.performCommand(BUILD_COMMAND);
-    theiaTerminal.waitTerminalOutput(EXPECTED_TERMINAL_OUTPUT, 1);
+    try {
+      theiaTerminal.waitTerminalOutput(EXPECTED_TERMINAL_OUTPUT, 1);
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      if (isTextPresentInTerminalOutput(EXPECTED_TERMINAL_SUCCESS_OUTPUT, 1)) {
+        fail("Known permanent failure https://github.com/eclipse/che/issues/11624", ex);
+      }
+
+      throw ex;
+    }
+
     theiaTerminal.waitTerminalOutput(EXPECTED_TERMINAL_SUCCESS_OUTPUT, 1);
   }
 
+  private boolean isTextPresentInTerminalOutput(String expectedText, int terminalIndex) {
```

Comment:
This method could be useful for other tests. What about moving it to the `TheiaTerminal` page object?